backends/tfhe-cuda-backend/cuda/src/integer/cmux.cuh

```diff
@@ -89,6 +89,17 @@ __host__ void host_integer_radix_cmux_kb(
                 mem_ptr->tmp_false_ct, params.big_lwe_dimension,
                 num_radix_blocks);
 
+  auto message_extract_lut_f = [params](Torus x) -> Torus {
+    return x % params.message_modulus;
+  };
+  generate_device_accumulator<Torus>(
+      streams[0], gpu_indexes[0],
+      mem_ptr->message_extract_lut->get_lut(gpu_indexes[0], 0),
+      params.glwe_dimension, params.polynomial_size, params.message_modulus,
+      params.carry_modulus, message_extract_lut_f);
+  mem_ptr->message_extract_lut->broadcast_lut(streams, gpu_indexes,
```

Comment:
The number of synchronizations in `broadcast_lut()` can impact performance here. Maybe we can implement a lazy sync by introducing a `sync_streams` parameter in that function. In cases like this we could synchronize all the streams only once inside `integer_radix_apply_univariate_lookup_table_kb`. Wdyt?

Reply:
Actually do we need to synchronize streams in broadcast_lut?

Reply:
Ok it seems that we do need it :sweat_smile:

Reply:
Good point. Probably not since any kernel dependency will be added to the same stream as the copy. We have to test to be sure, but we probably would be safe.

Reply:
I'll open an issue to do that in another PR, I'd like to merge this one asap :sweat_smile:

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/664